frontend/src/components/website/adding_master/StepAddDomain.vue

```diff
@@ -0,0 +1,127 @@
+                <VForm
+                    ref="form"
+                    lazy-validation
+                    @submit.prevent="onGoToNextStep"
+                >
+                    <VCardText class="mb-4 ps-0 mt-2 pt-0">
+                        <VTextField
+                            v-model="domain"
+                            label="Website domain"
+                            class="form-input"
+                            required
+                            single-line
+                            solo
+                            :error="!!errorText"
+                            :error-messages="errorText"
+                            :rules="domainRules"
+                        />
+                    </VCardText>
+                    <VSwitch
+                        v-model="single_page"
+                        label="Single page app"
+                        color="#3C57DE"
+                        inset
+                    />
+                </VForm>
+                <VBtn
+                    large
+                    class="white--text mt-4"
+                    color="#3C57DE"
+                    @click="onGoToNextStep"
+                >
+                    Get Tracking Info
+                </VBtn>
+            </VContainer>
+        </VCard>
+    </VContainer>
+</template>
+
+<script>
+    import {mapActions, mapGetters} from 'vuex';
+    import {GET_NEW_WEBSITE} from "@/store/modules/website/types/getters";
+    import {SAVE_NEW_WEBSITE, SET_DOMAIN, SET_SPA} from "@/store/modules/website/types/actions";
+
+    const domainRegex = /(https?:\/\/)?(www\.)?[-a-zA-Z0-9@:%._~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_.~#()?&//=]*)/;
+
+    export default {
+        name: 'StepAddDomain',
+        props: {
+            errorMessage: {
+                type: String,
+                required: false,
+                default: ''
+            }
+        },
+        data() {
+            return {
+                domain: '',
+                valid: false,
+                single_page: false,
+                domainRules: [
+                    v => !!v || 'Website domain is required',
+                    v => domainRegex.test(v) || 'Domain must be correct'
+                ],
+            }
+        },
+        computed: {
+            ...mapGetters('website', {
+                newWebsite: GET_NEW_WEBSITE
+            }),
+            errorText() {
+                return this.errorMessage;
+            }
+        },
+        created () {
+            this.domain = this.newWebsite.domain;
+            this.single_page = this.newWebsite.single_page;
+        },
+        methods: {
+            ...mapActions('website', {
+                setDomain: SET_DOMAIN,
+                setSinglePage: SET_SPA,
+                saveNewSite: SAVE_NEW_WEBSITE
+            }),
+            onGoToNextStep () {
+                if (this.$refs.form.validate()) {
+                    this.setDomain(this.domain).then(() => {
```

Comment:
no need to define action per field, you can define 1 action and 1 mutation, like `SET_DOMAIN_INFO` to update multiple fields state, and SAVE_ACTION - for saving, and call them like
```
this.setDomainInfo(...).then(() => this.saveNewSite()).then(this.onSaved).catch(this.onError);
```
where `onSaved` - success behavior, and `onError` - error behavior